### ISSUE TITLE
Unexpected New Tab

### ISSUE THREAD

**Issue by zzzeid** [mozilla-mobile/firefox-ios#10477]
## Steps to reproduce

1. Open a shortcut from the home page
2. Navigate to home screen (swipe up from the bottom)
3. Wait a few seconds (10s)
4. Reopen Firefox (either by clicking on the icon in the homescreen OR by navigating to existing open window via swiping up)

### Expected behavior
I should see the tab I was at before leaving Firefox.

### Actual behavior
A new tab opens. I can find where I was before by going to my tabs.

### Device & build information
* Device: iPhone 12 mini, iOS 15.3.1
* Build version: Firefox Daylight 98.2 (8475)

### Notes
This is not the only time this unexpected behaviour occurs. Sometimes it happens when I navigate to a link from another app -- and unexpectedly end up in a new tab.

A similar issue started with "jump back in", which behaved in a similar unexpected way.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-4092)

**Comment by lmarceau** [comment]
@zzzeid If you go in Settings > Homepage > Opening screen. Which options do you have in the following: `Homepage`, `Last tab` or `Homepage after four hours of inactivity`?

**Comment by zzzeid** [comment]
@lmarceau it shows `Homepage` checked. The description below it also clearly states "Choose what you see when you return to Firefox." which makes sense, since both workflows I mentioned are technically "returning" to Firefox. However, I believe this behaviour is different from what was happening a few months ago, and this change was a little unexpected / I was not aware of this setting. There's a possibly related workflow that is a little more difficult to reproduce, where a new link I click on sometimes ends up imperceptibly opening but immediately covered with a new tab. There also seems to be a delay of a few seconds of what "returning" actually means.

**Comment by zzzeid** [comment]
@lmarceau thanks for clarifying. It's definitely possible that I manually changed this setting in the past when I was fiddling with these settings. I upgraded and tried to reproduce the other bug (navigating from a link), but it appears to be fixed now.

On a side note, the newer Firefox version did not show up in the App Store until I manually searched for Firefox and clicked on it -- at which point the "open" button changed to "update". Looking at the list of available updates just before this did not show Firefox as needing an update. Not sure if this is a bug or something specific to my Phone but thought I would mention it.

In any case, I think this issue can be closed!
